types/result.go

```diff
@@ -207,6 +208,10 @@ func (r TxResponse) String() string {
 	if r.Timestamp != "" {
 		sb.WriteString(fmt.Sprintf("  Timestamp: %s\n", r.Timestamp))
 	}
+	if len(r.Events) > 0 {
+		bz, _ := cdc.MarshalJSON(r.Events)
```

Comment:
cdc is Amino, right? Maybe let's use `codec.ProtoMarshalJSON` instead

Reply:
Yes, we use `cdc` to JSON marshal the logs so I used that. Happy to change this to Proto.

Reply:
`r.Events` does not implement the `proto.Message` interface, so we cannot use `codec.ProtoMarshalJSON`.

Reply:
Ah true.

BTW, aren't we willing to move away from custom string representations that look like semi-YAML? For example, this String() implementation could simply be:

```go
return codec.MarshalYAML(codec.NewProtoCodec(nil), r)
```

where https://github.com/cosmos/cosmos-sdk/blob/bf11b1bf1fa0c52fb2cd51e4f4ab0c90a4dd38a0/codec/yaml.go#L11.

Similarly to what we're doing with BaseAccount

Reply:
Yes totally, unless of course the type needs custom YAML marshaling. Is there anything you want me to do here in context of this PR?

Reply:
I updated the `String` methods.

Reply:
I was thinking replacing the whole block L174-L215 by one line `return codec.MarshalYAML(codec.NewProtoCodec(nil), r)`. Would that work?

Reply:
Ahhh, yes. I think we can do that.